src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -128,16 +131,22 @@ public void Dispose()
             }
 
 #if DEBUG
+            private readonly string _creationCallStack;
+#endif
             ~Connection()
             {
                 // this can happen if someone kills OOP. 
                 // when that happen, we don't want to crash VS, so this is debug only check
                 if (!Environment.HasShutdownStarted)
                 {
-                    Contract.Requires(false, $@"Should have been disposed!");
+#if DEBUG
```

Comment:
:memo: If this had wrapped the entire finalizer from the start, this bug would be much less serious.

Reply:
not sure what you meant.